### PR TITLE
Fixed `yii\helpers\Inflector::camel2words()` to work with UTF-8

### PR DESCRIPTION
**Pull request by SilverFire**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | this one

### CODE REVIEW

**Review by samdark — APPROVED** [review]

**Comment by SilverFire** [comment]
There are many uses of non-unicode-safe functions (strtolower, ucwords, ucfirst) in inflector.
I think it makes sense to make them consistently unicode safe.

File: framework/helpers/BaseInflector.php [hunk 1]
```diff
@@ -369,13 +369,13 @@ public static function camelize($word)
...
-        $label = strtolower(trim(str_replace([
+        $label = mb_strtolower(trim(str_replace([
             '-',
             '_',
             '.',
-        ], ' ', preg_replace('/(?<![A-Z])[A-Z]/', ' \0', $name))));
+        ], ' ', preg_replace('/(\p{Lu})/u', ' \0', $name))));
 
-        return $ucwords ? ucwords($label) : $label;
+        return $ucwords ? mb_convert_case($label, MB_CASE_TITLE, 'UTF-8') : $label;
```
Comment (samdark) [inline framework/helpers/BaseInflector.php]: Yii::$app->charset?

**Comment by SilverFire** [comment]
Updated PR

File: framework/helpers/BaseInflector.php [hunk 2]
```diff
@@ -599,4 +600,27 @@ public static function sentence(array $words, $twoWordsConnector = null, $lastWo
...
+    }
+
+    /**
+     * The same as built-in `ucfirst`, but unicode-safe
+     *
+     * @param $string
+     * @param $encoding
+     * @return string
+     */
+    private static function mb_ucfirst($string, $encoding)
```
Comment (samdark) [inline framework/helpers/BaseInflector.php]: Could be renamed to `ucfirst`, made public and moved to `StringHelper`.

Comment (samdark) [reply framework/helpers/BaseInflector.php]: But overall I'm OK leaving it private. Parameter phpdoc types should be specified.

**Pull request merged**